Makefile

```diff
@@ -14,7 +14,67 @@ build:
 	'$(VIRTUALENV_DIR)/bin/pip' install -r ./requirements.txt
 	'$(VIRTUALENV_DIR)/bin/python' setup.py install
 
+# Create a image with docker build & docker run.
+docker-build:
+	# e.g., docker build -t  <image_name> /path/to/Dockerfile/
+	# -t flag - to tag the image with a name so Docker recognizes this name.
+	docker build -t migrate2to3:v1 .
+
+docker-run: docker-build
+	# Runs docker-build part before docker-run part, i.e., docker-run DEPENDS on docker-build.
+	docker run -it migrate2to3:v1 bash
+
+# Create 2 virtualenvs. Test that Python 2.7 & Python 3.6 is running correctly
+# within each virtualenv respectively.
+# PYTHON 2.7 TESTS.
+test-help-tablesnap-py2.7: docker-build
+	docker run migrate2to3:v1 ./py27env/bin/python ./tablesnap --help
+
+test-help-tableslurp-py2.7: docker-build
+	docker run -e USER=root migrate2to3:v1 ./py27env/bin/python ./tableslurp --help
+
+test-help-tablechop-py2.7: docker-build
+	docker run migrate2to3:v1 ./py27env/bin/python ./tablechop --help
+
+test-help-tablesnap-sunset-py2.7: docker-build
+	docker run migrate2to3:v1 ./py27env/bin/python ./tablesnap-sunset --help
+
+test-help-tablesnap-sync-manifest-py2.7: docker-build
+	docker run migrate2to3:v1 ./py27env/bin/python ./tablesnap-sync-manifest --help
+
+# # PYTHON 3.6 TESTS.
+test-help-tablesnap-py3.6: docker-build
+	docker run migrate2to3:v1 ./py36env/bin/python ./tablesnap --help
+
+test-help-tableslurp-py3.6: docker-build
+	docker run -e USER=root migrate2to3:v1 ./py36env/bin/python ./tableslurp --help
+
+test-help-tablechop-py3.6: docker-build
+	docker run migrate2to3:v1 ./py36env/bin/python ./tablechop --help
+
+test-help-tablesnap-sunset-py3.6: docker-build
+	docker run migrate2to3:v1 ./py36env/bin/python ./tablesnap-sunset --help
+
+test-help-tablesnap-sync-manifest-py3.6: docker-build
+	docker run migrate2to3:v1 ./py36env/bin/python ./tablesnap-sync-manifest --help
+
+DIRECTORY=/tmp/tablesnap-test
+TIMEOUT_A=5
+TIMEOUT_B=10
+AWS_ACC_KEY_ID := $(shell aws configure get aws_access_key_id)
+AWS_SEC_ACC_KEY := $(shell aws configure get aws_secret_access_key)
+BUCKET=$(USER)-tablesnap-tests
+HARDLINKS_PATH=/tmp/tablesnap-hardlinks
+
+docker-test-tablesnap: docker-build
+	docker run -it migrate2to3:v1 ./test_tablesnap.sh $(DIRECTORY) $(TIMEOUT_A) $(TIMEOUT_B) $(AWS_ACC_KEY_ID) $(AWS_SEC_ACC_KEY) $(BUCKET) $(HARDLINKS_PATH)
```

Comment:
For all the other tests you have a Python 2 version of the test, and a Python 3 version of the test, but for this test you use only Python 3. I suggest adding another test that runs the `test_tablesnap.sh` script but that uses Python 2. You should be able to do this by adding another parameter to the `test_tablesnap.sh` script that tells it whether to use Python 2 or Python 3. By default it could use Python 3.

In the `test_tablesnap.sh` script, you'd use that parameter to decide whether to use `py27env/bin/python` (Python 2) or `py36env/bin/python` to run the `tablesnap` program. If you'd like, we can discuss it looking at the files.

Reply:
Made changes for parameter of the Python version passed in from the Makefile in test_tablesnap.sh & Makefile.